latin_word_syllabification.py

```diff
@@ -0,0 +1,337 @@
+"""
+Module provides a function, syllabify_word, that syllabifies a Latin word. Although
+this function does not assume any specific spelling convention, it was 
+developed for use syllabifying texts (with both standard and non-standard spellings)
+entered in Cantus Database. As such, familiarity with the conventions of Cantus
+Database may be helpful; details can be found in the "Text Entry and Editing" pdf
+at https://www.cantusdatabase.org/documents). Consult the README for more
+details about the syllabification logic implemented in this function.
+
+syllabify_word(word, return_syllabified_string = False) 
+    word [str]: string to syllabify
+    return_syllabified_string [bool]: if True, returns a string with
+        syllables separated by hyphens; if False, returns a list of
+        indices of syllable boundaries. Defaults to False.
+
+returns a list of syllables with syllable boundaries marked by hyphens.
+
+The function split_word_by_syl_bounds can be used to split a word
+into syllables based on syllable boundaries. See function docstring
+for more details.
+
+Logs at level = DEBUG.
+"""
+
+import itertools
+import logging
+import re
+from typing import Union
+
+# Consonant groups are groups of consonants that are treated as a single
+# consonant for the purposes of syllabification. For details, see README.
+_CONSONANT_GROUPS: list = ["ch", "ph", "th", "rh", "gn", "qu", "gu", "nc", "mp", "sc"] + [
+    x[0] + x[1] for x in itertools.product("pbtdcfg", "lr")
```

Comment:
wait, *should* tl and dl be a consonant cluster? (I can't come up with many situations where they would arise, except some word like "quodlibet" which is surely quod-li-bet and not quo-dli-bet. But it's so rare I don't think it would matter much one way or the other.)

Reply:
Hmm....I got that part of the list from the syllabifier that Cantus Ultimus was using (https://github.com/frmatthew/exsurge/blob/f828578b2fc4f501697414080d4241a322ae3cc0/src/Exsurge.Text.js#L187) but now that you mention it, these cases do seem wrong. I've gone back through and convinced myself of all the non-`tl` and non-`dl` ones, but let me know if there are others.